### ISSUE TITLE
Async main fn is not detected

### ISSUE THREAD

**Issue by nate-onesignal** [integer32llc/rust-playground#563]
When using tokio, it is common to write an `async main` fn wrapped with the `#[tokio::main]` attribute. The playground fails to recognize the generated main fn and issues this warning:

```
No main function was detected, so your code was compiled
but not run. If you’d like to execute your code, please
add a main function.
```

Here is a basic example that fails to run:

```rust
#[tokio::main]
async fn main() {
    println!("Hello");
}
```

**Comment by shepmaster** [comment]
Thanks for filing this. I noticed it earlier but neglected to log it. 

To address this, [the regex detecting the `main` function needs to be updated](https://github.com/integer32llc/rust-playground/blob/5438d93f682f746e00c16fb4953c6c69d3b6c488/ui/frontend/selectors/index.ts#L13-L14).

**Comment by shepmaster** [comment]
Closed via #564
